tests/test_loaders.py

```diff
@@ -466,3 +466,66 @@ def test_multitracks():
         assert jam.validate(), "Jams validation failed for {}.MultiTrack({})".format(
             dataset_name, mtrack_id
         )
+
+
+def test_random_splits():
+    for dataset_name in DATASETS:
+        dataset = mirdata.initialize(
+            dataset_name, os.path.join(TEST_DATA_HOME, dataset_name), version="test"
+        )
+
+        list_sum_up_1 = [
+            [0.7, 0.1, 0.1, 0.1],
+            [0.4, 0.2, 0.2, 0.2],
+            [0.5, 0.2, 0.2, 0.1],
+            [0.8, 0.1, 0.1],
+            [0.7, 0.2, 0.1],
+            [0.9, 0.1, 0.0],
+            [0.6, 0.3, 0.1],
+            [0.5, 0.4, 0.1],
+            [0.9, 0.05, 0.05],
+            [0.8, 0.2],
+            [0.1, 0.9],
+        ]
+
+        list_not_sum_up_1 = [
+            [0.8, 0.1, 0.3, 0.2],
+            [0.3, 0.1, 0.3, 0.5],
+            [0.8, 0.1, 0.3],
+            [0.3, 0.1, 0.3],
+            [0.9, 0.2, 0.3],
+            [0.1, 0.1, 0.1],
+            [0.95, 0.01, 0.01],
+            [0.8, 0.1, 0.3],
+            [0.8, 0.3],
+            [0.1, 0.7],
+        ]
+
+        # check wrong type of split function
+        if dataset._track_class is None:
+            for right_combination in list_sum_up_1:
+                with pytest.raises(AttributeError):
+                    dataset.get_tracks_splits(right_combination)
+
+        if dataset._multitrack_class is None:
+            for right_combination in list_sum_up_1:
+                with pytest.raises(AttributeError):
+                    dataset.get_mtracks_splits(right_combination)
+
+        # check splits for tracks
+        if dataset._track_class:
+            for right_combination in list_sum_up_1:
+                splits = dataset.get_tracks_splits(right_combination)
+                assert len(dataset.track_ids) == sum([len(i) for i in splits])
+            for wrong_combination in list_not_sum_up_1:
+                with pytest.raises(ValueError):
+                    dataset.get_tracks_splits(wrong_combination)
+
+        # check splits for multitracks
+        if dataset._multitrack_class:
+            for right_combination in list_sum_up_1:
+                splits = dataset.get_mtracks_splits(right_combination)
+                assert len(dataset.mtrack_ids) == sum([len(i) for i in splits])
+            for wrong_combination in list_not_sum_up_1:
+                with pytest.raises(ValueError):
+                    dataset.get_mtracks_splits(wrong_combination)
```

Comment:
```suggestion
def test_random_splits():
    split = [0.9, 0.1]
    for dataset_name in DATASETS:
        dataset = mirdata.initialize(
            dataset_name, os.path.join(TEST_DATA_HOME, dataset_name), version="test"
        )
        
        # check wrong type of split function
        if dataset._track_class is None:
            with pytest.raises(AttributeError):
                dataset.get_track_splits(split)

        if dataset._multitrack_class is None:
            with pytest.raises(AttributeError):
                dataset.get_mtrack_splits(split)

        # check splits for tracks
        if dataset._track_class:
            splits = dataset.get_tracks_splits(split)
            assert len(dataset.track_ids) == sum([len(i) for i in splits])

        # check splits for multitracks
        if dataset._multitrack_class:
                splits = dataset.get_mtracks_splits(split)
                assert len(dataset.mtrack_ids) == sum([len(i) for i in splits])

```

This is a great test! I think it's actually doing much more than we need so here's my suggestion to simplify this test, and a suggested test to add to `test_core`. There's no way in github for me to directly suggest this, so here's my suggested function to add to `tests/test_core.py`:

```python
def test_dataset_splits():
    empty_dataset = core.Dataset()
    
    # test the case where there are no tracks
    with pytest.raises(AttributeError):
        empty_dataset.get_track_splits([0.9, 0.1])
    
    # test the case where there are no multitracks
    with pytest.raises(AttributeError):
        empty_dataset.get_mtrack_splits([0.9, 0.1])

    # test the partition function
    items = [i for i in range(100)]
    list_sum_up_1 = [
        [0.7, 0.1, 0.1, 0.1],
        [0.4, 0.2, 0.2, 0.2],
        [0.5, 0.2, 0.2, 0.1],
        [0.8, 0.1, 0.1],
        [0.7, 0.2, 0.1],
        [0.9, 0.1, 0.0],
        [0.6, 0.3, 0.1],
        [0.5, 0.4, 0.1],
        [0.9, 0.05, 0.05],
        [0.8, 0.2],
        [0.1, 0.9],
    ]
    for right_combination in list_sum_up_1:
        splits = empty_dataset._get_partitions(items, right_combination, 42)
        # check that the right number of splits are created
        assert len(splits) == len(right_combination)
        # check that the number of total items matches
        assert len(items) == sum([len(i) for i in splits])
        # check that all items are used
        assert set(items) == set([i for split_items in splits for i in split_items])
        # check that splits are nonoverlapping
        used = set()
        for split in splits:
            this_split = set(split)
            assert not this_split.intersection(used)
            used.update(this_split)
        # check that the split is reproducable
        splits2 = empty_dataset._get_partitions(items, right_combination, 42)
        assert splits == splits2
        
    list_not_sum_up_1 = [
        [0.8, 0.1, 0.3, 0.2],
        [0.3, 0.1, 0.3, 0.5],
        [0.8, 0.1, 0.3],
        [0.3, 0.1, 0.3],
        [0.9, 0.2, 0.3],
        [0.1, 0.1, 0.1],
        [0.95, 0.01, 0.01],
        [0.8, 0.1, 0.3],
        [0.8, 0.3],
        [0.1, 0.7],
    ]
    for wrong_combination in list_not_sum_up_1:
        with pytest.raises(ValueError):
            empty_dataset._get_partitions(items, wrong_combination, 42)

```

Reply:
Great! I will add this test to `test_core` during this week.

Reply:
The line `empty_dataset = core.Dataset()` is giving me problems, any suggestions?

```python
>       empty_dataset = core.Dataset()

tests/test_core.py:729: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
mirdata/core.py:116: in __init__
    self.data_home = self.default_path if data_home is None else data_home
mirdata/core.py:193: in default_path
    return os.path.join(mir_datasets_dir, self.name)
/usr/lib/python3.8/posixpath.py:90: in join
    genericpath._check_arg_types('join', a, *p)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

funcname = 'join', args = ('/home/miguel/mir_datasets', None), hasstr = True, hasbytes = False, s = None

    def _check_arg_types(funcname, *args):
        hasstr = hasbytes = False
        for s in args:
            if isinstance(s, str):
                hasstr = True
            elif isinstance(s, bytes):
                hasbytes = True
            else:
>               raise TypeError(f'{funcname}() argument must be str, bytes, or '
                                f'os.PathLike object, not {s.__class__.__name__!r}') from None
E               TypeError: join() argument must be str, bytes, or os.PathLike object, not 'NoneType'

/usr/lib/python3.8/genericpath.py:152: TypeError
```

Reply:
Ah my bad! Try:

Reply:
Still not working 😥
I also tried `empty_dataset = core.Dataset(name="empty_test", indexes={"default": 0})`
And then I'm getting `AttributeError: 'int' object has no attribute 'get_path'` not sure how the inner part of `core.Dataset` works.

Reply:
looking into it and will get back to you shortly!

Reply:
@migperfer I just updated the suggested test for test_core - It works for me off of your branch. Let me know if you run into anything though

Reply:
✨✨Looks like it's working!